scripts/fast-reboot

```diff
@@ -370,7 +370,7 @@ function check_docker_exec()
 {
     containers="radv bgp lldp swss database teamd syncd"
     for container in $containers; do
-        STATE=$(timeout 1s docker exec $container echo "success"; if [[ $? == 124 ]]; then echo "timed out"; fi)
+        STATE=$(timeout 5s docker exec $container echo "success"; if [[ $? == 124 ]]; then echo "timed out"; fi)
```

Comment:
I am not in favor of this change. IMO, 1s should be long enough for docker exec to respond.

This change is motivated by limitations of test infra/test. The 1s check is to validate if docker exec won't be hung in later stage. 5s test here seem too generous to the extent that the need for this check itself is nullified.

Reply:
If in your test "hardware simulation platform" takes too much of CPU, then you could modify this timeout as part of test (ugly patch at runtime). But changing image's behavior for test limitation is not right.

Reply:
@vaibhavhd I can image this happening on a real hardware with lower end CPU and/or under heavy load. I observe docker command sometimes hang for a moment, even on real device or even on our server. Seems to be a dockerd thing. The "docker exec" is communicating with dockerd and makes it a limiting factor IMO. E.g, we have processes running in the background and executing docker commands. procdockerstatsd is one such example. There might be more such background processes creating a load on dockerd. Thus, IMO, timeout 1 sec is a concern even on a real hardware.

If the timeout is there just to make sure it won't hang forever it is better to increase to be on a safe side. There's nothing wrong if docker exec takes 2, 10 sec, especially if it is not running at the end of this script that it could affect control plane restoration time. I see the choice of 1 sec is arbitrary, other commands have 5 sec, so I adjusted this line to other places in the script.

Reply:
Stepan, if the delay is experienced on device's CPU/hardware then it is in best interest to not proceed with fast/warm reboot. This is due to the fact that this slowness can degrade the timing requirements in both shutdown and boot up path. In which case, we will likely see dataplane disruption due to different reasons, such as LAG flap.

The timeout not only prevents a hung state, but also detect a degraded state.

Reply:
@yxieca, any thoughts here?

Reply:
I agree. If docker exec takes 5 seconds to complete, I have serious doubt on if the warm shutdown / recovery can be done within protocol time window.